### PR TITLE
Bugfix/numpy dtype deprecation

### PR DESCRIPTION
**Pull request by sbernasek**
# Citrine Python PR

## Description 
Changes pandas dependency requirement to Pandas>=1.1.0 in order to avoid numpy dtype alias deprecation warnings thrown by pandas<=1.1 when NumPy>=1.20.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -9,3 +9,4 @@ boto3==1.17.93
 botocore==1.20.95
 deprecation==2.0.7
 urllib3==1.26.5
+pandas>=1.1.0
```
Comment (maxhutch) [inline requirements.txt:12]: Why has this been added?

**Review by maxhutch — CHANGES_REQUESTED** [review]
Shouldn't we bump the lower bound in [setup.py](https://github.com/CitrineInformatics/citrine-python/blob/main/setup.py#L35) as well?

File: requirements.txt (hunk 1 above)
Comment (sbernasek) [reply requirements.txt:12]: I hadn't seen that we were handling pandas versioning via `extras_require` in setup.py and was intending to shield users from the same numpy vs pandas version discrepancy that was causing my local tests to fail, but I suppose that's outside the scope of our requirements. I'll drop this requirement and upgrade the optional dependency to 1.1

Comment (sbernasek) [reply requirements.txt:12]: Alternatively, if we don't care about the warning and just want to ensure that our tests pass, we can leave the user requirement for pandas as is (1.0.03 as a builders dependency in setup.py) and just change the test requirement.

Comment (maxhutch) [reply requirements.txt:12]: I think we should bump the lower bound in the builders dependency and leave it out of here.

Comment (sbernasek) [reply requirements.txt]: Ok, I dropped the pandas dependency in requirements.txt, bumped the lower bound on the builders dependency, and left the test_requirements version as 1.1.0.

**Review by maxhutch — APPROVED** [review]

**Pull request merged**